packages/core/src/payment/strategies/braintree/braintree-ach-payment-strategy.ts

```diff
@@ -0,0 +1,184 @@
+import { PaymentMethodFailedError } from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import {
+    InvalidArgumentError,
+    MissingDataError,
+    MissingDataErrorType,
+    NotInitializedError,
+    NotInitializedErrorType,
+} from '../../../common/error/errors';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { PaymentArgumentInvalidError } from '../../errors';
+import { UsBankAccountInstrument } from '../../payment';
+import PaymentActionCreator from '../../payment-action-creator';
+import PaymentMethod from '../../payment-method';
+import PaymentMethodActionCreator from '../../payment-method-action-creator';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import { BraintreeError, BraintreeUsBankAccount, UsBankAccountSuccessPayload } from './braintree';
+import BraintreePaymentProcessor from './braintree-payment-processor';
+import isBraintreeError from './is-braintree-error';
+import isUsBankAccountInstrumentLike from './is-us-bank-account-instrument-like';
+
+export default class BraintreeAchPaymentStrategy implements PaymentStrategy {
+    private _paymentMethod?: PaymentMethod;
+    private _usBankAccount?: BraintreeUsBankAccount;
+    private _mandateText = '';
+
+    constructor(
+        private _store: CheckoutStore,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _braintreePaymentProcessor: BraintreePaymentProcessor,
+    ) {}
+
+    async initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { braintreeach } = options;
+
+        if (!options.methodId) {
+            throw new InvalidArgumentError(
+                'Unable to initialize payment because "options.methodId" argument is not provided.',
+            );
+        }
+
+        if (!braintreeach) {
+            throw new InvalidArgumentError(
+                `Unable to initialize payment because "options.braintreeach" argument is not provided.`,
+            );
+        }
+
+        if (!braintreeach?.mandateText) {
+            throw new InvalidArgumentError(
+                'Unable to initialize payment because "options.braintreeach.mandateText" argument is not provided.',
+            );
+        }
+
+        this._mandateText = braintreeach?.mandateText;
+
+        const state = await this._store.dispatch(
```

Comment:
Maybe better use try catch in case if something will go wrong

Reply:
This expression has been removed